test-assets/ffbc54/shortcut.js

```diff
@@ -0,0 +1,34 @@
+const defaultParams = {
+  target: 'target',
+  focusOnly: false, 
+  shortcutKey: '+', 
+  ctrlKey: false,
+  disabled: false
+};
+
+let allSettings = new Array();
+
+function shortcut(...params) {
+  allSettings = [...allSettings, ...params.map(p => { return { ...defaultParams, ...p } })];
```

Comment:
I think we can make that much more readable by using `allSettings.push(…)` which will update the array directly and be more clear on the fact that we are just adding values to an array.
As a small benefit, `push()` actually changes the content of the array, but not the array itself, so `allSettings` could be made a `const`…

---
This function is actually doing two things: (i) registering a new setting into `allSettings`; and (ii) activating all shortcuts defined in `allSettings` (the `addEventListener` part). I think they can be separated. Given that the settings are read from `allSettings` when the event is triggered (not when it is registered) this part can be an independant function that is called only once in each `onload` attribute. Then, I would go as far as having the register part only take one setting at a time. That way, this code would be super simple:
```javascript
function registerSetting(param) {
  allSettings.push({...defaultParams, ...params})
}
```
and the examples would do something like `onload="registerSetting({ /*shortcut 1*/}); registerSetting({ /*shortcut 2*/ }); activateShortcuts()"`.

---
Additionally, it becomes obvious that way that each element in `allSettings` is actually the parameters for a shortcut definition… So it should probably be named `allShorstcuts` or `shortcutsDefinitions` or something similar (and `registerShortcut` is a very explicit name also).